### PR TITLE
Update `@grafana/data` to 8.5.4

### LINKED ISSUE #28: Opensearch: Replace angular annotation editor with react editor
Remove angular annotation editor and replace it with react annotation editor.

### PR DESCRIPTION
**Pull request by svennergr**
In order to fix #28 the dependencies - especially those belonging to `@grafana` - needed to be updated.
I also updated further core dependencies such as react or jest. 

After updating I had some problems with running tests, which were fixed by the changes in jest.config.js.

### CODE REVIEW

**Comment by svennergr** [comment]
The CI shows that the dependency `react-dev-utils` needs Node v14, but it seems like plugins need Node v12.
How do we handle those cases? 

Reasoning for upgrading `react-dev-utils` were some vulnerabilities inside that dependency [[Source PR](https://github.com/grafana/grafana/pull/44696)].

**Comment by joey-grafana** [comment]
Hmm maybe we can leave updating react-dev-utils for another PR - to keep this one more focused.

I also see that both `"react-dev-utils": "^12.0.0"` & `"@types/react-dev-utils": "9.0.10"` are dependencies in grafana

**Comment by joey-grafana** [comment]
May need to revert updating some other libraries too as build_frontend is failing.

**Comment by joey-grafana** [comment]
This is fantastic, great job on getting the build fixed. Will take another look at this after lunch :)

**Comment by svennergr** [comment]
Hi @joey-grafana 

I now also updated `@grafana/toolkit` to version 8.5.4. This will come with a few benefits:
- all `@grafana` dependencies are running on the same version
- plugin is working with those updates (important benefit, right? 😅)

However, since this plugin should also work with Node 12 I needed to pin the following dependencies to a version which is fine for Node 12:
- "react-dev-utils": "10.2.1",
- "@typescript-eslint/eslint-plugin": "4.0.1",
- "@typescript-eslint/parser": "4.0.1"

What do you think about this?

**Comment by joey-grafana** [comment]
> Hi @joey-grafana
> 
> I now also updated `@grafana/toolkit` to version 8.5.4. This will come with a few benefits:
> 
> * all `@grafana` dependencies are running on the same version
> * plugin is working with those updates (important benefit, right? 😅)
> 
> However, since this plugin should also work with Node 12 I needed to pin the following dependencies to a version which is fine for Node 12:
> 
> * "react-dev-utils": "10.2.1",
> * "@typescript-eslint/eslint-plugin": "4.0.1",
> * "@typescript-eslint/parser": "4.0.1"
> 
> What do you think about this?

That's good! Please double check that there are no console errors after running yarn install, then yarn watch :) You can also check out elasticsearch to see how it deals with the newer versions where the package api has changed and requires updates in opensearch.

**Comment by svennergr** [comment]
> > Hi @joey-grafana
> > I now also updated `@grafana/toolkit` to version 8.5.4. This will come with a few benefits:
> > 
> > * all `@grafana` dependencies are running on the same version
> > * plugin is working with those updates (important benefit, right? 😅)
> > 
> > However, since this plugin should also work with Node 12 I needed to pin the following dependencies to a version which is fine for Node 12:
> > 
> > * "react-dev-utils": "10.2.1",
> > * "@typescript-eslint/eslint-plugin": "4.0.1",
> > * "@typescript-eslint/parser": "4.0.1"
> > 
> > What do you think about this?
> 
> That's good! Please double check that there are no console errors after running yarn install, then yarn watch :) You can also check out elasticsearch to see how it deals with the newer versions where the package api has changed and requires updates in opensearch.

I like the idea of just updating `@grafana/data` much better. Did that now, which seems to work after arranging some types.

File: src/components/QueryEditor/QueryTypeEditor/index.tsx [hunk 1]
```diff
@@ -7,10 +7,14 @@ import { queryTypeConfig, getQueryTypeOptions } from './utils';
 import { segmentStyles } from '../styles';
 import { QueryType } from '../../../types';
 
-const toOption = (queryType: QueryType) => ({
-  label: queryTypeConfig[queryType].label,
-  value: queryType,
-});
+const toOption = (queryType: QueryType) => {
+  console.log('queryType :>> ', queryType);
+  console.log('queryTypeConfig', queryTypeConfig);
```
Comment (joey-grafana) [inline src/components/QueryEditor/QueryTypeEditor/index.tsx:12]: Please remove

**Review by joey-grafana — APPROVED** [review]
LGTM - great job Sven 💯

File: src/components/QueryEditor/QueryTypeEditor/index.tsx (hunk 1 above)
Comment (svennergr) [reply src/components/QueryEditor/QueryTypeEditor/index.tsx]: Sorry, missed them. Removed now!

**Pull request merged**